mythtv/libs/libmythbase/compat.h

```diff
@@ -335,7 +353,7 @@ static __inline struct tm *localtime_r(const time_t *timep, struct tm *result)
 #    define ftello(stream) ftello64(stream)
 #endif
 
-#if defined(USING_MINGW) && defined(FILENAME_MAX)
+#if defined(USING_MINGW) && defined(FILENAME_MAX) && defined(__cplusplus)
```

Comment:
Use a separate block to exclude only the C++ headers if not C++
dirent.h and readdir_ are still relevant for C

Reply:
This should be unnecessary. There aren't any C files left in the code base outside of libmythmpeg2, and nothing in that directory includes compat.h.

Reply:
Removed this inline function as it is no longer required for mingw.